src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -6880,6 +6880,59 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                 value.m_value = args[2].m_value;
                 fn_args.push_back(al, value);
             }
+        } else if(attr_name == "center") {
+            if (args.size() != 1 && args.size() != 2) {
+                throw SemanticError("str.center() takes one or two argument",
+                    loc);
+            }
+            ASR::expr_t *arg_value = args[0].m_value;
+            ASR::ttype_t *arg_value_type = ASRUtils::expr_type(arg_value);
+            if (!ASRUtils::is_integer(*arg_value_type)) {
+                throw SemanticError("str.center() argument 1 must be integer", loc);
+            }
+
+            fn_call_name = "_lpython_str_center";
+            ASR::call_arg_t str;
+            str.loc = loc;
+            str.m_value = s_var;
+
+            ASR::call_arg_t value;
+            value.loc = loc;
+            value.m_value = args[0].m_value;
+            fn_args.push_back(al, str);
+            fn_args.push_back(al, value);
+
+            if(args.size()==2){
+                ASR::expr_t *arg_value = args[1].m_value;
+                ASR::ttype_t *arg_value_type = ASRUtils::expr_type(arg_value);
+                if (!ASRUtils::is_character(*arg_value_type)) {
+                    throw SemanticError("str.center() argument 2 must be str", loc);
+                }
+                value.m_value = args[1].m_value;
+                fn_args.push_back(al, value);
+            }
+        } else if(attr_name == "expandtabs") {
+            if(args.size() > 1) {
+                throw SemanticError("str.expandtabs() takes at most one argument.", loc);
+            }
+            fn_call_name = "_lpython_str_expandtabs";
+            ASR::call_arg_t str;
+            str.loc = loc;
+            str.m_value = s_var;
+            fn_args.push_back(al, str);
+
+            if(args.size()==1){
+                ASR::expr_t *arg_value = args[0].m_value;
+                ASR::ttype_t *arg_value_type = ASRUtils::expr_type(arg_value);
+                if (!ASRUtils::is_integer(*arg_value_type)) {
+                    throw SemanticError("str.expandtabs() argument must be integer", loc);
+                }
+
+                ASR::call_arg_t value;
+                value.loc = loc;
+                value.m_value = args[0].m_value;
+                fn_args.push_back(al, value);
+            }
```

Comment:
This handles runtime implementation. We also need to support compile-time implementation inside https://github.com/lcompilers/lpython/blob/7d0cfdbec3269b3aaaeea57a012e7b9135194ba1/src/lpython/semantics/python_ast_to_asr.cpp#L6909.